### ISSUE TITLE
Some nodes unavailable for scheduling

### ISSUE THREAD

**Issue by counterbeing** [kontena/kontena#1630]
Here you can see I have three nodes that are currently available for scheduling containers.
![image](https://cloud.githubusercontent.com/assets/239798/21758613/4a87cfc6-d5f2-11e6-8222-abf7b986d6a9.png)

But, if I look at the list of nodes, I see all five.

```
➜  rails_app git:(gem_update) ✗ kon node ls

Name                                                                   Status     Initial    Labels
misty-bush-85                                                          online     yes        region=sfo1,az=sfo1,provider=digitalocean
black-snow-30                                                          online     yes        region=sfo1,az=sfo1,provider=digitalocean
long-firefly-44                                                        online     yes        region=sfo1,az=sfo1,provider=digitalocean
late-water-57                                                          online     no         -
proud-sun-11                                                           online     no         region=sfo1,az=sfo1,provider=digitalocean
```

I first saw one node popping on and off. I had no idea why, but I decided to try removing it and adding another to take its place. While removing it, another node dropped off. But, the new node joined and started scheduling as expected. Now the nodes list reports that all nodes are online, but they never show up when I'm looking at the monitor. 

I just SSH'd into one of the missing nodes and was greeted by this: 

```
Last login: Mon Jan  9 06:28:33 UTC 2017 from 50.0.193.51 on pts/0
Container Linux by CoreOS stable (1235.4.0)
Failed Units: 3
  systemd-coredump@1-24873-0.service
  systemd-coredump@2-24883-0.service
  systemd-coredump@3-24897-0.service
```

This is on a system provisioned by the digital ocean plugin, with no manual modifications. 

And, when I'm on the host and check the running containers, I see containers that I expect to be running.  I've got about 300Mb of free memory. 

A few notes about context: 
- I was playing with scaling up and down  around the time that this happened.
- I just updated to 1.0.4 today.

Please let me know if any service logs in particular would help and i'll pull them.

**Comment by counterbeing** [comment]
I've been tinkering with this for a bit. I tried rebooting nodes initially. Suddenly one of them popped back up. I removed and replaced it. Everything seems to be running fine now. I don't know what happened. I would love to understand a bit better, just to help me trust the system.

**Comment by counterbeing** [comment]
I managed to reproduce this again. I tried to scale down my my web service using `kontena service scale rails_app/web 7` and all of a sudden, all of my nodes disappear. As mentioned above, I was also tinkering with the scale command when the problem originally occurred.

Here's are some snippets from the logs as i'm not entirely sure what piece is the problem:

```
weave | INFO: 2017/01/10 03:52:05.465750 Discovered local MAC 26:67:18:c5:01:a3
weave | INFO: 2017/01/10 03:52:05.459417 Discovered remote MAC 26:67:18:c5:01:a3 at 52:2b:96:f2:4c:87(silent-wind-84)
weave | INFO: 2017/01/10 03:52:05.460508 Discovered remote MAC 26:67:18:c5:01:a3 at 52:2b:96:f2:4c:87(silent-wind-84)
weave | INFO: 2017/01/10 03:52:05.462994 Discovered remote MAC 26:67:18:c5:01:a3 at 52:2b:96:f2:4c:87(silent-wind-84)
weave | INFO: 2017/01/10 03:52:05.458226 Discovered remote MAC 26:67:18:c5:01:a3 at 52:2b:96:f2:4c:87(silent-wind-84)
kontena-cadvisor | E0110 03:52:05.482415    2430 fsHandler.go:106] failed to collect filesystem stats - du command failed on /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a with output stdout: 53524	/rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a
kontena-cadvisor | , stderr: du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/1640/fd/104: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/1640/fdinfo/109: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/2206/fdinfo/41: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/3350/fdinfo/115: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/3471/fdinfo/115: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/3608/fdinfo/92: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/1435/task/3808/fd/92: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/8239: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/8245: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/8252: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/8262: No such file or directory
kontena-cadvisor | du: /rootfs/var/lib/docker/overlay/722ac85b904d4dd91b94e8a7ba5df38be74d1db2e66d4e3e0cd38a54e975930a/merged/proc/8274: No such file or directory
weave | INFO: 2017/01/10 03:52:05.552526 [nameserver 52:2b:96:f2:4c:87] adding entry for 41f08e5a8f5f2a54b340377b4194540186d8080f1204f66b28e7534fb7d9d77c: web-9.rails_app.production.kontena.local. -> 10.81.128.77
kontena-agent | I, [2017-01-10T03:52:36.944142 #1]  INFO -- Kontena::Workers::NodeInfoWorker: publishing node information
kontena-agent | I, [2017-01-10T03:52:43.087114 #1]  INFO -- Kontena::Workers::StatsWorker: starting collection
```

**Comment by counterbeing** [comment]
I decided to try restarting nodes again. After restarting all of them, they register as online after about 30 seconds, then about 2-3 minutes later they go back to offline.

```
Name                                                                   Status     Initial    Labels
misty-bush-85                                                          offline    yes        region=sfo1,az=sfo1,provider=digitalocean
black-snow-30                                                          offline    yes        region=sfo1,az=sfo1,provider=digitalocean
silent-wind-84                                                         offline    yes        -
late-water-57                                                          offline    no         -
hidden-snow-91                                                         offline    no         region=sfo1,az=sfo1,provider=digitalocean
```

**Comment by SpComb** [comment]
TBH I'm not exactly sure how the agent errors (systemd coredumps and the cadvisor errors) relate, but the last part about the nodes being online for a minute or two does sound very much like #1626, which is about error handling on the server side.

**Comment by SpComb** [comment]
Can you SSH in to your kontena-master node and take a look at `docker logs kontena-master-api`, if it's logging anything interesting when or before this happens?

**Comment by counterbeing** [comment]
@SpComb Thanks for the response. I actually hopped on slack, and some Jussi helped me get up and running by rebooting the master. I left my logs there in general. And there's a bit of conversation around all of this. The master log didn't seem particularly interesting, but rebooting the master did bring my nodes back.

**Comment by SpComb** [comment]
Closed by #1626, which will let the master log an error and restart in these kinds of situations.

Re-open/comment if you still see similar behavior or interesting error messages and server restarts in similar situations with 1.0.5 once released.
